### ISSUE TITLE
Node running out of Heap Memory

### ISSUE THREAD

**Issue by gerrysweeney** [vuejs/vitepress#1460]
### Discussed in https://github.com/vuejs/vitepress/discussions/1458

<div type='discussions-op-text'>

<sup>Originally posted by **gerrysweeney** October 10, 2022</sup>
I have been trying to use Vitepress to generate documentation, but it fails on build with a Node.js "Out of Heap Memory" error. It seems to keep everything in memory while building, it never creates the dist folder, consumes over 4Gb of memory and then I get the attached error.  Any idea how I can resolve that? and/or can the build process be made more memory efficient?

<img width="1424" alt="Screenshot 2022-10-10 at 20 47 31" src="https://user-images.githubusercontent.com/9318317/194942287-be08280c-3778-41e6-88a5-0b7a7d2e5315.png">
</div>

**Comment by brc-dd** [comment]
Hi! Can you provide some reproduction? It should not happen. For me, it's taking only a few hundred MBs. Also, passing `max_old_space_size` like this won't work. You're passing it to `vitepress` not `node` here. Try something like this instead:

```sh
cross-env NODE_OPTIONS=--max-old-space-size=3584 vitepress build docs
```

Also please attach the output of

```sh
npx envinfo --system --npmPackages vitepress --binaries --browsers
```

**Comment by akil-rails** [comment]
I am having the same issue, have already set NODE_OPTIONS=--max_old_space_size=4096, and it worked for sometime, and now it fails again.

**Comment by brc-dd** [comment]
Can you tell the size (in MB) of your projects? I am guessing you're hitting https://github.com/vitejs/vite/issues/2433 here.

**Comment by kiaking** [comment]
Closing due to inactivity.

**Comment by brc-dd** [comment]
fixed in #3366
